### PR TITLE
Ratings models and admin

### PR DESCRIPTION
**Pull request by Par010**
- docstring changes for courses admin
- added a ratings app
- models for ratings 
- admin for ratings 
- derived attributes for average rating per attribute per resource

### CODE REVIEW

File: ratings/models.py [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+
+        # calculates the average of the qs list
+        average = sum(qs)/float(len(qs))
+
+        # rounds off the average to the closest 0.5
+        result = round(average * 2) / 2.0
+        return result
+
+    @property
+    def get_avg_attribute2(self):
```
Comment (darth-dodo) [inline ratings/models.py]: how to dry this out? this is very inefficient.

File: ratings/models.py [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+        """
+        qs = KnowledgeBaseRating.objects.filter(resource=self.resource).values_list('attribute_4', flat=True).order_by('id')
+        average = sum(qs)/float(len(qs))
+        result = round(average * 2) / 2.0
+        return result
+
+    class Meta:
+
+        # user and resource ids together cannot be duplicated
+        unique_together = ('user', 'resource')
```
[the remark this answers is not in the document]
Comment (Par010) [reply ratings/models.py]: It does allow changing the values of ratings but does not allow duplicate entry with the same user and resource

[the remark this answers is not in the document]
Comment (darth-dodo) [reply ratings/models.py]: but what about one resource being shared across a lot of "pathways"? let's remove the constraint for now. we will think about how to map pathways to rating in S3

File: courses/models.py [hunk 3]
```diff
@@ -196,6 +196,15 @@ def save(self, *args, **kwargs):
             self.slug = pl_custom_slugify(source_field=self.title, suffix=True)
             super(SoftSkillsData, self).save(*args, **kwargs)
 
+    @property
+    def ratings(self):
```
Comment (darth-dodo) [inline courses/models.py]: why not a model manager?

**Pull request merged**